GeekbenchSwift/fft.swift

```diff
@@ -1,33 +1,77 @@
 // Copyright (c) 2014 Primate Labs Inc.
 // Use of this source code is governed by the 2-clause BSD license that
 // can be found in the LICENSE file.
-
 import Foundation
 
+struct Complex : Printable {
+    internal var real : Float32
+    internal var imaginary : Float32
+    var description: String {
+        return String(format: "(%.5f, %.5fi)", real, imaginary)
+    }
+    
+    init() {
+        self.real = 0
+        self.imaginary = 0
+    }
+    
+    init(real : Float32, imaginary : Float32) {
+        self.real = real
+        self.imaginary = imaginary
+    }
+    
+    mutating func assign(real : Float32, imaginary : Float32) {
+        self.real = real
+        self.imaginary = imaginary
+    }
+    
+    mutating func assign(rhs : Complex) {
+        self.real = rhs.real
+        self.imaginary = rhs.imaginary
+    }
+    
+    mutating func add(rhs : Complex) {
+        self.real += rhs.real
+        self.imaginary += rhs.imaginary
+    }
+    
+    mutating func subtract(rhs : Complex) {
+        self.real -= rhs.real
+        self.imaginary -= rhs.imaginary
+    }
+}
+
+func * (left : Complex, right : Complex) -> Complex {
+    return Complex(real: left.real * right.real - left.imaginary * right.imaginary,
+        imaginary: left.real * right.imaginary + left.imaginary * right.real)
+}
+
+func + (left : Complex, right : Complex) -> Complex {
+    return Complex(real: left.real + right.real, imaginary: left.imaginary + right.imaginary)
+}
+
+func - (left : Complex, right : Complex) -> Complex {
+    return Complex(real: left.real - right.real, imaginary: left.imaginary - right.imaginary)
+}
+
+
 final class SFFTWorkload : Workload {
   let pi = Float32(acos(-1.0))
 
-  var size : Int
-  var chunkSize : Int
-  var input : [Complex] = []
-  var output : [Complex] = []
+  let size : Int
+  let chunkSize : Int
+  var input : [Complex]
+  var output : UnsafeMutablePointer<Complex>
   var wFactors : [Complex] = []
 
   init(size : Int, chunkSize : Int) {
     self.size = size
     self.chunkSize = chunkSize
 
-    self.input.reserveCapacity(size)
-
-    for _ in 0..<size {
-      self.input.append(Complex())
-    }
-
-    self.output.reserveCapacity(chunkSize)
-
-    for _ in 0..<chunkSize {
-      self.output.append(Complex())
-    }
+    self.input = [Complex](count: size, repeatedValue: Complex())
+    self.output = UnsafeMutablePointer<Complex>.alloc(size)
```

Comment:
`output` can be of size `chunkSize` instead of `size` since we overwrite it for each chunk.